source/onednngraph.js

```diff
@@ -0,0 +1,515 @@
+
+    constructor(metadata, symbol, version) {
+        if (!symbol) {
+            throw new onednngraph.Error('JSON symbol data not available.');
+        }
+        if (symbol) {
+            if (!Object.prototype.hasOwnProperty.call(symbol, 'graph')) {
+                throw new onednngraph.Error('JSON file does not contain an oneDNN Graph \'graph\' property.');
+            }
+            if (!Object.prototype.hasOwnProperty.call(symbol, 'version')) {
+                throw new onednngraph.Error('JSON file does not contain an oneDNN Graph \'version\' property.');
+            }
+        }
+        this._format = 'oneDNN Graph';
+        this._version = 'oneDNN' + (version ? ' v' + version : '');
+        this._license = 'Apache License 2.0';
+        this._graphs = [new onednngraph.Graph(metadata, symbol)];
+    }
+
+    get format() {
+        return this._format;
+    }
+
+    get producer() {
+        return this._producer;
+    }
+
+    get name() {
+        return this._name;
+    }
+
+    get version() {
+        return this._version;
+    }
+
+    get description() {
+        return this._description;
+    }
+
+    get author() {
+        return this._author;
+    }
+
+    get license() {
+        return this._license;
+    }
+
+    get runtime() {
+        return this._runtime;
+    }
+
+    get graphs() {
+        return this._graphs;
+    }
+};
+
+onednngraph.Graph = class {
+
+    constructor(metadata, symbol) {
+        this._metadata = metadata;
+        this._nodes = [];
+        this._inputs = [];
+        this._outputs = [];
+
+        if (symbol) {
+            this._type = 'engine: ' + symbol.engine_kind + '; fpmath: ' + symbol.fpmath_mode + ';';
+            const nodes = symbol.graph;
+
+            const args = new Map();
+            const arg = (name, type, lt) => {
+                if (!args.has(name)) {
+                    args.set(name, new onednngraph.Argument(name, type, null, lt));
+                }
+                return args.get(name);
+            };
+            for (const node of nodes) {
+                this._nodes.push(new onednngraph.Node(this._metadata, node, arg, symbol.engine_kind));
+            }
+        }
+    }
+
+    get name() {
+        return '';
+    }
+
+    get type() {
+        return this._type;
+    }
+
+    get inputs() {
+        return this._inputs;
+    }
+
+    get outputs() {
+        return this._outputs;
+    }
+
+    get nodes() {
+        return this._nodes;
+    }
+};
+
+onednngraph.Node = class {
+
+    constructor(metadata, node, arg, device) {
+        const type = node.kind;
+        this._name = node.name;
+        this._attributes = [];
+        this._inputs = [];
+        this._outputs = [];
+        this._type = { name: type };
+        this._device = device;
+        this._description = 'oneDNN Graph OP with unique id: ' + node.id;
+
+        const attrs = node.attrs;
+        if (attrs) {
+            for (const attributeName of Object.keys(attrs)) {
```

Comment:
`Object.entries` instead of `Object.keys` to avoid duplicate lookups?

Reply:
good catch.